src/extension/coqPilot.ts

```diff
@@ -81,70 +75,126 @@ export class CoqPilot {
             this.performCompletionForAllAdmits.bind(this)
         );
 
+        const toggleCommand = `toggle_current_session`;
+        this.registerEditorCommand(
+            toggleCommand,
+            this.toggleCurrentSession.bind(this)
+        );
+        this.healthStatusIndicator = new HealthStatusIndicator(
+            `${pluginId}.${toggleCommand}`,
+            vscodeExtensionContext
+        );
+        this.healthStatusIndicator.updateStatusBar(true);
+
         this.vscodeExtensionContext.subscriptions.push(this);
     }
 
+    static async create(vscodeExtensionContext: ExtensionContext) {
+        const globalExtensionState = new GlobalExtensionState();
+        const sessionExtensionState = await SessionExtensionState.create(
+            globalExtensionState.logOutputChannel,
+            globalExtensionState.eventLogger
+        );
+        return new CoqPilot(
+            vscodeExtensionContext,
+            globalExtensionState,
+            sessionExtensionState
+        );
+    }
+
     async performCompletionUnderCursor(editor: TextEditor) {
         const cursorPosition = editor.selection.active;
         this.performSpecificCompletionsWithProgress(
             (hole) => positionInRange(cursorPosition, hole.range),
-            editor
+            editor,
+            this.sessionExtensionState.abortController.signal
         );
     }
 
     async performCompletionInSelection(editor: TextEditor) {
         const selection = editor.selection;
         this.performSpecificCompletionsWithProgress(
             (hole) => selection.contains(toVSCodePosition(hole.range.start)),
-            editor
+            editor,
+            this.sessionExtensionState.abortController.signal
         );
     }
 
     async performCompletionForAllAdmits(editor: TextEditor) {
-        this.performSpecificCompletionsWithProgress((_hole) => true, editor);
+        this.performSpecificCompletionsWithProgress(
+            (_hole) => true,
+            editor,
+            this.sessionExtensionState.abortController.signal
+        );
+    }
+
+    async toggleCurrentSession() {
+        if (this.globalExtensionState.hasActiveSession) {
```

Comment:
Maybe it's better to move the activity status to the session state? Then the global state might work again as a context that only stores some data. Additionally, it's better to minimize dependencies between any components/interfaces

Reply:
And yes, if we do so, then the `SessionState` itself will be responsible for setting its activity indicator to false after the `abort` / `dispose` call

Reply:
Done. Everything until handling the updates of UI in SessionState was easy and neat. But accurately passing the StatusBar to the SessionState was kind of tricky in terms of dependencies. I hope its fine rn